hadml/datamodules/gan_datamodule.py

```diff
@@ -251,3 +250,141 @@ def test_dataloader(self):
             ),
         }
         return CombinedLoader(loaders, mode="max_size_cycle")
+
+
+class MultiHadronEventGANDataModule(LightningDataModule):
+    def __init__(
+            self, 
+            data_dir="data/Herwig",
+            raw_file_list=["AllClusters_10K.dat"],
+            processed_filename="herwig_multihadron_events_10K.npy",
+            pid_map_file="pid_to_idx.pkl",
+            train_val_test_split: Tuple[float, float, float] = (0.7, 0.15, 0.15),
+            batch_size=32,
+            num_workers=8,
+            initialise_data_preparation=False,
+            debug=True
+        ):
+        super().__init__()
+        self.processed_filename = os.path.join(os.path.normpath(data_dir),
+                                               "processed", processed_filename)
+        self.train_val_test_split = train_val_test_split
+        self.batch_size = batch_size
+        self.num_workers = num_workers
+        self.data_train: Optional[torch.utils.data.Dataset] = None
```

Comment:
It's not important to fix, but just so you know: Typically the dataset is imported directly - `from torch.utils.data import Dataset`. As it is now, it takes up a lot of characters whenever used

Reply:
In general, I think most of torch classes are typically either on the top-level (`torch.sth`), or a submodule/class is imported from torch before use.